CARDS-2609: Configurable actions on user dashboard and related views

acrowthe opened the pull request:
Implement configurable clinician views for the dashboard, form and subject lists
- Adds actionSwitches which allow buttons and other actions to be enabled or disabled on these views
- Adds options to existing views and buttons to support different URLs extensions enabling them to be reused between different paths (admin vs. non-admin)
- Add support for customizing the columns shown in FormViews

To be built and run alongside [SPARC-36](https://github.com/data-team-uhn/sparc/pull/47)

[SPARC-36]: https://phenotips.atlassian.net/browse/SPARC-36?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

acrowthe committed:
CARDS-2609: Configurable actions on user dashboard and related views
Add admin option to current clinician pages

acrowthe committed:
CARDS-2609: Configurable actions on user dashboard and related views
Finish new clinician views

acrowthe committed:
CARDS-2609: Configurable actions on user dashboard and related views
Finish clinician view implementation

acrowthe committed:
CARDS-2609: Configurable actions on user dashboard and related views
Improve form table UI

sdumitriu commented on modules/data-entry/src/main/frontend/src/dataHomepage/LiveTable.jsx:
```diff
@@ -45,7 +45,7 @@ function LiveTable(props) {
 
   const { customUrl, resourceSelectors, columns, showTotalRows, defaultLimit, updateData, classes,
-    filters, entryType, actions, admin, disableTopPagination, disableBottomPagination,
+    filters, entryType, actions, admin, extensionURL, disableTopPagination, disableBottomPagination,
```
`admin` isn't used anymore, can it be removed?

sdumitriu commented on modules/data-entry/src/main/frontend/src/dataHomepage/FormView.jsx:
```diff
@@ -44,13 +44,16 @@ import NewFormDialog from "./NewFormDialog.jsx";
   const [ filtersJsonString, setFiltersJsonString ] = useState(new URLSearchParams(window.location.hash.substring(1)).get("forms:filters"));
 
+  const extensionURL = extension?.["cards:extensionURL"] || props.extensionURL || ""
+  const baseURL = "/content.html" + extensionURL ? "/" + extensionURL : ""
```
Bug, `?` has lower priority than `+`, so the condition is always true and thus `baseURL = "/"`

sdumitriu committed:
[cleanup] Remove no longer valid "admin" props

sdumitriu committed:
[misc] Fix double // in some URLs

sdumitriu committed:
CARDS-2609: Configurable actions on user dashboard and related views
Consistently pass the extension definition down to the actual extension component
